probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -762,6 +967,16 @@ where
         Ok(())
     }
 
+    fn write_16(&mut self, address: u64, data: &[u16]) -> Result<(), ArmError> {
+        if data.len() == 1 {
```

Comment:
Since the inherent `write_16` is public, do you think this branch should have a better place there?

Reply:
Since this was already done this way for 8bit and 32bit reads, I would prefer doing this in a separate PR to avoid this one getting too big. 

(I think it should be possible to make everything call the inherent `read_N` methods to simplify, these seem to be equally efficient than the `read_word_N` methods)